test/e2e/performanceprofile/functests/10_mixedcpus/mixedcpus.go

```diff
@@ -0,0 +1,160 @@
+		})
+
+		It("should update runtime configuration with the given shared cpuset", func() {
+			workers, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			// test arbitrary one should be good enough
+			worker := &workers[0]
+			cmd := []string{
+				"chroot",
+				"/rootfs",
+				"/bin/bash",
+				"-c",
+				fmt.Sprintf("/bin/grep -E 'shared_cpuset.*' %s | awk -F '\"' '{ print $2 }'", crioRuntimeConfigFile),
+			}
+			cpus, err := nodes.ExecCommandOnNode(cmd, worker)
+			Expect(err).ToNot(HaveOccurred(), "failed to execute command on node; cmd=%q node=%q", cmd, worker)
+			cpus = strings.Trim(cpus, "\n")
+			crioShared, err := cpuset.Parse(cpus)
+			Expect(err).ToNot(HaveOccurred())
+			// don't need to check the error, the values were already validated.
+			shared, _ := cpuset.Parse(string(*profile.Spec.CPU.Shared))
+			Expect(shared.Equals(crioShared)).To(BeTrue(), "crio config file does not contain the expected shared cpuset; shared=%q crioShared=%q",
+				shared.String(), crioShared.String())
+		})
+	})
+})
+
+func setup(ctx context.Context, profile *performancev2.PerformanceProfile) func(ctx context.Context) {
+	initialProfile := profile.DeepCopy()
+	isolated, err := cpuset.Parse(string(*profile.Spec.CPU.Isolated))
+	Expect(err).ToNot(HaveOccurred())
+
+	// arbitrary take the first one
+	sharedcpu := cpuset.New(isolated.List()[0])
+	testlog.Infof("shared cpu ids are: %q", sharedcpu.String())
+	updatedIsolated := isolated.Difference(sharedcpu)
+	profile.Spec.CPU.Isolated = cpuSetToPerformanceCPUSet(&updatedIsolated)
+	profile.Spec.CPU.Shared = cpuSetToPerformanceCPUSet(&sharedcpu)
+	profile.Spec.WorkloadHints.MixedCpus = pointer.Bool(true)
+	Expect(testclient.Client.Update(ctx, profile)).ToNot(HaveOccurred(), "failed to update profile: %q", profile.Name)
+
+	mcp, err := mcps.GetByProfile(profile)
+	Expect(err).ToNot(HaveOccurred())
+
+	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+	teardown := func(ctx context.Context) {
+		Expect(testclient.Client.Get(ctx, client.ObjectKeyFromObject(initialProfile), profile))
+		profile.Spec = initialProfile.Spec
+		resourceVersion := profile.ResourceVersion
+		Expect(testclient.Client.Update(ctx, profile)).ToNot(HaveOccurred(), "failed to update profile: %q", profile.Name)
+		// do not wait if nothing has changed
+		if resourceVersion != profile.ResourceVersion {
+			mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		}
+	}
+	return teardown
+}
+
+func cpuSetToPerformanceCPUSet(set *cpuset.CPUSet) *performancev2.CPUSet {
+	c := performancev2.CPUSet(set.String())
+	return &c
+}
+
+func isFileExistCmd(absoluteFileName string) []string {
```

Comment:
looks correct. A stronger test could be try to read the file and ensure it has _any_ content on it. Is not much costlier than the current test. But it's debatable we need it on this PR, so let's save for the future.

Reply:
I replace `-f` with `-s` to check if file exist and not empty. Thanks!